usr/lib/common/pkey_utils.c

```diff
@@ -602,3 +702,343 @@ CK_RV pkey_aes_cmac(OBJECT *key_obj, CK_BYTE *message,
 
     return ret;
 }
+
+/**
+ * Determine the CPACF curve type from given template.
+ */
+cpacf_curve_type_t get_cpacf_curve_type(TEMPLATE *tmpl)
+{
+    const CK_BYTE p256[] = OCK_PRIME256V1;
+    const CK_BYTE p384[] = OCK_SECP384R1;
+    const CK_BYTE p521[] = OCK_SECP521R1;
+    CK_BYTE *ec_params;
+    CK_ULONG ec_params_len;
+    CK_ATTRIBUTE *attr = NULL;
+    cpacf_curve_type_t curve_type;
+
+    /* Determine CKA_EC_PARAMS */
+    if (template_attribute_get_non_empty(tmpl, CKA_EC_PARAMS, &attr) != CKR_OK) {
+        TRACE_ERROR("%s: This template does not have CKA_EC_PARAMS, cannot determine curve type.\n",
+                    __func__);
+        curve_type = curve_invalid;
+        goto done;
+    }
+    ec_params = (CK_BYTE *) attr->pValue;
+    ec_params_len = attr->ulValueLen;
+
+    if (memcmp(ec_params, p256, ec_params_len) == 0)
```

Comment:
Comparison only makes sense if `ec_params_len == sizeof(p256)`.  Otherwise we might end up reading parts of other variables (or curves).  Same for cases below.

Reply:
Done.